src/panels/lovelace/editor/card-editor/hui-card-picker.ts

```diff
@@ -74,38 +278,172 @@ export class HuiCardPicker extends LitElement {
-        .cards-container mwc-button {
-          flex: 1 0 25%;
+
+        .card {
+          width: calc(33% - 8px);
           margin: 4px;
+          cursor: pointer;
+        }
+
+        @media all and (max-width: 780px), all and (max-height: 500px) {
+          .card {
+            flex: 1 0 100%;
+          }
         }
 
-        @media all and (max-width: 450px), all and (max-height: 500px) {
-          .cards-container mwc-button {
-            flex: 1 0 33%;
+        @media all and (max-width: 1200px), all and (max-height: 500px) {
+          .card {
+            width: calc(50% - 8px);
           }
         }
+
+        ha-card {
+          height: 100%;
+          display: flex;
+          flex-direction: column;
+        }
+
+        ha-card > :first-child {
+          pointer-events: none;
+          margin: 0 20px 20px 20px;
+        }
+
+        .options {
+          border-top: 1px solid #e8e8e8;
+          padding: 5px 8px;
+          display: flex;
+          margin-top: 8px;
+        }
+
+        .options .primary-actions {
+          flex: 1;
+          margin: auto;
+        }
+
+        .preview-text {
+          color: var(--disabled-text-color);
+          font-size: 18px;
+          margin-top: 0;
+        }
+
+        .preview > :nth-child(2) {
+          zoom: 0.6;
+          -moz-transform: scale(0.6); /* Firefox */
+          -moz-transform-origin: 0 0;
+          -o-transform: scale(0.6); /* Opera */
+          -o-transform-origin: 0 0;
+        }
       `,
     ];
   }
 
-  private _manualPicked(): void {
-    fireEvent(this, "config-changed", {
-      config: { type: "" },
-    });
+  private _cardPicked(ev: Event): void {
+    const config: LovelaceCardConfig = (ev.currentTarget! as CardPickTarget)
+      .config;
+
+    fireEvent(this, "config-changed", { config });
   }
 
-  private async _cardPicked(ev: Event): Promise<void> {
-    const type = (ev.currentTarget! as CardPickTarget).type;
+  private async _getCardConfig(
+    cardConfig: CardPickerConfig
+  ): Promise<LovelaceCardConfig | undefined> {
+    if (!this.hass) {
+      return undefined;
+    }
+
+    let config: LovelaceCardConfig = cardConfig.lovelaceCardConfig;
+
+    if (cardConfig.noEntity) {
+      return config;
+    }
 
-    const elClass = await getCardElementClass(type);
-    let config: LovelaceCardConfig = { type };
+    const elClass = await getCardElementClass(
+      cardConfig.lovelaceCardConfig.type
+    );
 
     if (elClass && elClass.getStubConfig) {
-      const cardConfig = elClass.getStubConfig(this.hass!);
-      config = { ...config, ...cardConfig };
+      const stubConfig = elClass.getStubConfig(this.hass!);
+      config = { ...stubConfig, ...config };
     }
 
-    fireEvent(this, "config-changed", { config });
+    return config;
+  }
+
+  private _getCardEntities(cardConfig: CardPickerConfig): string[] {
+    let entityIds: string[] =
+      this.entities || computeUnusedEntities(this.hass!, this.lovelace!);
```

Comment:
We shouldn't compute unused/used entities for every card type. Do it once and reuse it.